### PR TITLE
fix: fix an issue with rendering boolean hp values in learning curve table [#2670]

### PR DESCRIPTION
**Pull request by hamidzr**
## Test Plan

- manually tested with boolean hp values

## Checklist

- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: webui/react/src/pages/ExperimentDetails/ExperimentVisualization/HpTrialTable.tsx [hunk 1]
```diff
@@ -103,7 +103,7 @@ const HpTrialTable: React.FC<Props> = ({
         if (isNumber(value) && isValidType) {
           return <HumanReadableFloat num={value} />;
         }
-        return record.hparams[key];
```
Comment (hamidzr) [inline webui/react/src/pages/ExperimentDetails/ExperimentVisualization/HpTrialTable.tsx:106]: the confusion comes from `boolean` being a valid `JSX.Element` while it's a value doesn't get rendered at all 🤔

**Comment by hamidzr** [comment]
@hkang1 any ideas how we test this better than doing a manual test? 

maybe through a new story on our storybook?

**Review by hkang1 — APPROVED** [review]
LGTM thanks for

**Comment by hamidzr** [comment]
added a story for showcasing rendering of boolean hp values 

image

File: webui/react/src/pages/ExperimentDetails/ExperimentVisualization/HpTrialTable.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    experimentId={1}
+    hyperparameters={{ xyz: { type: ExperimentHyperParamType.Categorical, vals: [ true, false ] } }}
+    metric={{ name: 'metricA', type: MetricType.Training }}
+    trialHps={[
+      { hparams: { xyz: true }, id: 1, metric: 0.3 },
+      { hparams: { xyz: false }, id: 2, metric: 1.23 },
+    ]}
+    trialIds={[ 1, 2 ]}
+  />;
+};
```
Comment (hkang1) [inline webui/react/src/pages/ExperimentDetails/ExperimentVisualization/HpTrialTable.stories.tsx:26]: Awesome, thanks for adding this story!

**Pull request merged**